doc/building.rst

```diff
@@ -4,99 +4,188 @@
-recipes.
+2. Install Xcode command line tools:
 
-To set up your build environment for development using HomeBrew_:
+   .. code-block:: bash
 
-1. Install Xcode
-2. Install Xcode command line tools::
-    xcode-select --install
+      % xcode-select –install
 
-3. Install homebrew::
-    /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+3. Install Homebrew from `brew.sh`_ (which will just give you the following):
 
-4. Add the Nextcloud repository using the following command::
+.. _`brew.sh`: https://brew.sh
 
-    brew tap owncloud/owncloud
+   .. code-block:: bash
 
-5. Install a Qt5 version with qtwebkit support::
+      % /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
 
-    brew install qt5
+.. note:: Under certain cirumstances, you may get on error along the
+   lines of ``Permission denied @ apply2files`` when installing certain
+   Homebrew packages. This is `a known issue`_ and can be fixed by changing
+   the permissions on the affecte files with the following command:
 
-6. Install any missing dependencies::
+   .. code-block:: bash
 
-    brew install $(brew deps owncloud-client)
+      % sudo chown -R $(whoami):admin /usr/local/* \
+         && sudo chmod -R g+rwx /usr/local/*
 
-7. Add Qt from brew to the path::
+   This workaround may lead to other shell warnings.
 
-    export PATH=/usr/local/Cellar/qt5/5.x.y/bin:$PATH
+.. _`a known issue`: https://stackoverflow.com/a/63241724
 
-   Where ``x.y`` is the current version of Qt 5 that brew has installed
-   on your machine.
-8. Install qtkeychain from here:  git clone https://github.com/frankosterfeld/qtkeychain.git
-   make sure you make the same install prefix as later while building the client e.g.  -
-   ``DCMAKE_INSTALL_PREFIX=/Path/to/client-install``
+4. Install Homebrew packages:
 
-9. For compilation of the client, follow the :ref:`generic-build-instructions`.
+   .. code-block:: bash
 
-10. Install the Packages_ package creation tool.
+      % brew install git qt qtkeychain cmake openssl pkgconfig glib doxygen cmocka
 
-11. Enable git submodules:
-```
-$ cd desktop
-$ git submodule init
-$ git submodule update
-```
+5. Certain Homebrew packages are not automatically linked in places where
+   the build scripts can find them, so create a shell-profile script that
+   can load the necessary environmental variables when you run a build:
 
-12. Generate the build files:
-```
-$ cd build
-$ cmake .. -DCMAKE_INSTALL_PREFIX=~/nextcloud-desktop-client -DCMAKE_BUILD_TYPE=Debug
-```
+   .. code-block:: bash
 
-13. Compile and install:
-```
-$ make install
-```
+      % echo 'export OPENSSL_ROOT_DIR=$(brew --prefix openssl)' >> ~/.nextcloud_build_variables
+      % echo 'export QT_PATH=$(brew --prefix qt5)/bin' >> ~/.nextcloud_build_variables
+      % echo 'export Qt5LinguistTools_DIR=$(brew --prefix qt5)/lib/cmake/Qt5LinguistTools/' >> ~/.nextcloud_build_variables
```

Comment:
You seem to insist to not use `CMAKE_PREFIX_PATH`. Any reasons? :) But if this way works too it's also okay:)

Reply:
I might have done it wrong, but `CMAKE_PREFIX_PATH` didn't seem to work as an environmental variable (rather than a suffix to the `cmake` command). I haven't tried it again, though.

Reply:
I tried it again, and it worked.